src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +746,37 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	@Override
+	public void removeNodeFromParent(MutableTreeNode node) {
+		super.removeNodeFromParent(node);
+		publishEvent((SiteNode) node.getParent(), (SiteNode) node, EventType.REMOVE);
+	}
 
+	/**
+	 * Publishes the add or remove event. Node events are always published.
+	 * Site events are only published when the parent of the node is the root of the tree.
+	 * 
+	 * @param parent relevant parent node
+	 * @param node the site node the action is being carried out for
+	 * @param event the type of event occurring (ADD or REMOVE)
+	 * 
+	 * @see {@code EventType}
+	 * @since (TODO add version)
+	 */
+	
+	private void publishEvent(SiteNode parent, SiteNode node, EventType eventType) {
+		switch (eventType) {
+		case ADD:
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_NODE_ADDED_EVENT, new Target(node)));

```

Comment:
This statement(s) can be extracted to another method, one that just does:
```Java
ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), new Event(SiteMapEventPublisher.getPublisher(), event, new Target(node)));
```
We would just need to change one place, if anything needs to be changed in how the events are published.